### PR TITLE
Add integration tests from opm-simulators

### PR DESCRIPTION
**Pull request by hakonhagland**
Adds PYACTION integration tests from opm-simulators.

### CODE REVIEW

File: cmake/modules/OpmPythonAddEmbeddedTests.cmake [hunk 1]
```diff
@@ -1,16 +1,198 @@
...
     endforeach()
-    set_tests_properties(msim_ACTIONX PROPERTIES
-            ENVIRONMENT "PYTHONPATH=${PROJECT_BINARY_DIR}/opm-python/src:$ENV{PYTHONPATH}")
   endif()
 endfunction()
-add_opm_embedded_tests()
\ No newline at end of file
+
+#
+#  TODO: This function is part of the opm-simulators file compareECLFiles.cmake and should
```
Comment (lisajulia) [inline cmake/modules/OpmPythonAddEmbeddedTests.cmake:19]: Yes, ideally this is done before merging.

File: cmake/modules/OpmPythonAddTests.cmake [hunk 2]
```diff
@@ -1,7 +1,11 @@
+# TODO: Use the opm_add_test() macro in opm-common's OpmSatellites.cmake to add tests
```
Comment (lisajulia) [inline cmake/modules/OpmPythonAddTests.cmake:1]: and this as well.. :smiley:

File: cmake/modules/OpmPythonAddTests.cmake [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
+#    instead of using add_test() as below.
+
 add_test(NAME opmcommon_tests
        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/${GENERATED_PROJECT_DIR}/tests/opmcommon"
        COMMAND ${Python3_EXECUTABLE} ${PROJECT_SOURCE_DIR}/helper_scripts/discover_opmcommon_tests.py)
 set_tests_properties(opmcommon_tests PROPERTIES FIXTURES_REQUIRED CopyFilesFixture)
+#add_dependencies(test-suite "opmcommon_tests")
```
Comment (lisajulia) [inline cmake/modules/OpmPythonAddTests.cmake:8]: Can this commented out line be removed?

File: cmake/modules/OpmPythonAddTests.cmake [hunk 4]
```diff
@@ -15,4 +19,5 @@ foreach(case_name IN ITEMS basic fluidstate_variables primary_variables schedule
        -E env PYTHONPATH=${PROJECT_BINARY_DIR}/${GENERATED_PROJECT_DIR}/src:${PROJECT_BINARY_DIR}/${GENERATED_PROJECT_DIR}/tests
        ${Python3_EXECUTABLE} -m unittest opmsimulators/test_${case_name}.py)
   set_tests_properties(python_${case_name} PROPERTIES FIXTURES_REQUIRED CopyFilesFixture)
+  #add_dependencies(test-suite python_${case_name})
```
Comment (lisajulia) [inline cmake/modules/OpmPythonAddTests.cmake:22]: Can this commented out line be removed?

File: cmake/modules/OpmPythonAddEmbeddedTests.cmake [hunk 5]
```diff
@@ -1,16 +1,198 @@
-function (add_opm_embedded_tests)
+
+function (add_opm_embedded_tests_from_opm_common)
   if (Boost_UNIT_TEST_FRAMEWORK_FOUND)
     set(_libs mocksim opmcommon ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY})
 
-    foreach( test test_msim test_msim_ACTIONX test_msim_EXIT)
-      opm_add_test(${test} SOURCES embedded/tests/msim/${test}.cpp
+    foreach( test msim msim_ACTIONX msim_EXIT)
```
Comment (lisajulia) [inline cmake/modules/OpmPythonAddEmbeddedTests.cmake:6]: test can be removed, only `test_msim_ACTIONX.cpp  test_msim.cpp  test_msim_EXIT.cpp` are in the embedded/tests/msim folder

**Review by lisajulia — CHANGES_REQUESTED** [review]

File: cmake/modules/OpmPythonAddEmbeddedTests.cmake (hunk 1 above)
Comment (hakonhagland) [reply cmake/modules/OpmPythonAddEmbeddedTests.cmake:19]: Yes :)

File: cmake/modules/OpmPythonAddEmbeddedTests.cmake (hunk 5 above)
Comment (hakonhagland) [reply cmake/modules/OpmPythonAddEmbeddedTests.cmake:6]: `test` is here the loop variable, right? It is not a specific test case

Comment (lisajulia) [reply cmake/modules/OpmPythonAddEmbeddedTests.cmake:6]: ah yes, you're right :+1:

File: cmake/modules/OpmPythonAddTests.cmake (hunk 3 above)
Comment (lisajulia) [reply cmake/modules/OpmPythonAddTests.cmake:8]: ok

File: cmake/modules/OpmPythonAddEmbeddedTests.cmake (hunk 5 above)
Comment (hakonhagland) [reply cmake/modules/OpmPythonAddEmbeddedTests.cmake:6]: Thanks for the review :smile: I am merging this now.

**Pull request merged**